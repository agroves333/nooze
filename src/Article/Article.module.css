.article {
  color: #333;
  cursor: pointer;
  padding: 10px;
  transition: all .2s ease-in-out; }

.article:hover {
  text-decoration: none;
  background-color: #eee;
  transform: scale(1.05); }

.image {
  width: 100%; }
