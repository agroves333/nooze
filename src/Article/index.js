import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import get from 'lodash/get';

import styles from './Article.module.css';

const Article = ({ data }) => {
  const { headline, snippet, web_url, multimedia } = data;
  
  const image = multimedia && multimedia.find(item => {
    return item.subtype === 'blog427';
  });
  
  const imageUrl = get(image, 'url', '');

  return (
    <article className={classnames(styles.article, 'article')} onClick={() => window.open(web_url)}>
      {imageUrl && <img className={styles.image} src={`http://www.nytimes.com/${imageUrl}`} alt="Article image" />}
      <strong className={styles.title}>{headline.main}</strong>
      <div className={styles.summary}>{snippet}</div>
    </article>
  );
};

Article.propTypes = {
  data: PropTypes.shape({
    headline: PropTypes.shape({
      main: PropTypes.string,
    }),
    snippet: PropTypes.string,
    web_url: PropTypes.string,
    multimedia: PropTypes.array,
  }),
};

Article.defaultProps = {
  data: {
    headline: {
      main: '',
    },
    snippet: '',
    web_url: '',
    multimedia: [],
  }
};

export default Article;
