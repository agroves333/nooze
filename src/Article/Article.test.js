import React from 'react';
import { shallow } from 'enzyme';
import Article from './index';

const baseData = {
  headline: {
    main: 'fake-headline',
  },
  snippet: 'fake-snippet',
  web_url: 'fake-url',
  multimedia: [{
    subtype: 'blog427',
    url: 'fake-url',
  }],
};

describe('Article', () => {
  it('renders an Article', () => {
    const wrapper = shallow(<Article data={baseData} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('does not render image element if not available', () => {
    const wrapper = shallow(<Article data={{ ...baseData, multimedia: [] }} />);
    const image = wrapper.find('img');
    expect(image.exists()).toBe(false);
  });
  
  it('renders an image element if available', () => {
    const wrapper = shallow(<Article data={baseData} />);
    const image = wrapper.find('img');
    expect(image.exists()).toBe(true);
  });
  
  it('opens link when clicked', () => {
    global.open = jest.fn();
    const wrapper = shallow(<Article data={baseData}/>);
    wrapper.find('article').simulate('click');
    expect(global.open).toBeCalledWith('fake-url');
  });
});

