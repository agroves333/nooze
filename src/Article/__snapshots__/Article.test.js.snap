// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders an Article 1`] = `
<article
  className="article article"
  onClick={[Function]}
>
  <img
    alt="Article image"
    className="image"
    src="http://www.nytimes.com/fake-url"
  />
  <strong
    className="title"
  >
    fake-headline
  </strong>
  <div
    className="summary"
  >
    fake-snippet
  </div>
</article>
`;
